### PR TITLE
PR - Main Infra check

### LINKED ISSUE #897: 🐛 PR - Fix check to ensure main infra code works
**Describe the bug**
There is a failing check https://github.com/SSWConsulting/SSW.Website/actions/workflows/pr-infra.yml

```[tasklist]
### Tasks
- [x] Fix the title
- [x] Fix the triggers (change to PR)
- [x] Add the missing parameters
```

### PR DESCRIPTION
**Pull request by amankumarrr**
Fixes #897 

**Description:**
Removing the duplicate name of workflows, added testing for the main's BICEP with the pull request, and fixing the Service Principal ID error with the flow.

### CODE REVIEW

File: .github/workflows/pr-main-infra-check.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
-name: Main - Infra - Build and deploy
+name: Main - Infra Check - Build and deploy
```
Comment (bradystroud) [inline .github/workflows/pr-main-infra-check.yml]:
```suggestion
name: PR - Infra Check - Build and deploy
```

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-main-infra-check.yml (hunk 1 above)
Comment (amankumarrr) [reply .github/workflows/pr-main-infra-check.yml]: But we are checking main's infra there, do you think we should make these changes?

Comment (bradystroud) [reply .github/workflows/pr-main-infra-check.yml]: Looking at the other workflows, I think the naming scheme is {{ TRIGGER }} - {{ ACTION }}

Maybe a better name would be `PR - Test main infra`

Comment (amankumarrr) [reply .github/workflows/pr-main-infra-check.yml]: Yes it makes more sense.

**Review by bradystroud — APPROVED** [review]
lgtm

**Review by wicksipedia — APPROVED** [review]

**Pull request merged**